demos/machine_translation_demo/python/machine_translation_demo.py

```diff
@@ -204,18 +210,24 @@ def main(args):
     model = Translator(
         model_xml=args.model,
         model_bin=args.model.with_suffix(".bin"),
+        device=args.device,
         tokenizer_src=args.tokenizer_src,
         tokenizer_tgt=args.tokenizer_tgt,
         output_name=args.output_name
     )
-
-    if args.input:
-        def sentences():
-            for sentence in args.input:
+    input_data = args.input
+    if input_data and len(input_data) == 1 and input_data[0].endswith('.txt'):
+        input_data = open(input_data[0], 'r', encoding='utf8')
```

Comment:
This leaks the file handle. Please rewrite the code so that it doesn't happen.

(One simple way would be to just read the whole file at once.)

Reply:
Out of curiosity, won't python close the file when the object is destroyed?

Reply:
> Out of curiosity, won't python close the file when the object is destroyed?

Yes, but there is no guarantee for when the object is destroyed.

Reply:
On exit is also fine :)

Reply:
In this particular case it might be fine, but I think it's a good practice to ensure that any resources you allocate are freed when they are no longer needed. An idealistic reason is that the code might be reused in a different context where leaking the resource is not harmless (e.g. a long-running process). A more pragmatic reason is that Python will give you a `ResourceWarning` if you don't do it (visible in [development mode](https://docs.python.org/3/library/devmode.html)).

Reply:
I see